### PR TITLE
Add some docs on Py formatting

### PR DESCRIPTION
**Pull request by nilsvu**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tools/Hooks/pre-commit.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ done
 
 # Python file checks
 if [ ${#python_files[@]} -ne 0 ]; then
-    # Use black to check Python file formatting, only if it is installed.
+    # Use black to check Python file formatting
     @Python_EXECUTABLE@ -m black --version > /dev/null
     if [ $? -eq 0 ]; then
         @Python_EXECUTABLE@ -m black --check --quiet ${python_files[@]}
```
Comment (knelli2) [inline tools/Hooks/pre-commit.sh:53]: What do you think of actually running black here like we actually run `clang-format` on c++ files? Maybe also putting them in the diff that `ClangFormat.py` creates?

**Review by knelli2 — COMMENTED** [review]
What do you think about also doing this

```diff
diff --git a/docs/Installation/Installation.md b/docs/Installation/Installation.md
index 014cef7b70..b89fb99938 100644
--- a/docs/Installation/Installation.md
+++ b/docs/Installation/Installation.md
@@ -93,7 +93,7 @@ all of these dependencies.
   Alternatively, you can set `BOOTSTRAP_PY_DEPS=ON` when configuring a build
   with CMake to install missing Python packages into the build directory
   automatically.
-  <details>
+  <details open>
   \include support/Python/requirements.txt
   </details>
 
@@ -107,7 +107,7 @@ all of these dependencies.
   Install with `pip3 install -r support/Python/dev_requirements.txt`.
   Make sure you are working in a [Python venv](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment)
   before installing packages.
-  <details>
+  <details open>
   \include support/Python/dev_requirements.txt
   </details>
 * [Google Benchmark](https://github.com/google/benchmark) - to do
```

When I searched for hidden things on Chrome (linux), it actually opened the `Details` tab and highlighted the search. However, after some searching, apparently this behavior is browser dependent and not all browsers will search hidden content. So if we want to guarantee searchability in the installation docs, we can just have the details open initially (or just get rid of the hidden details and have the file included always).

File: tools/Hooks/pre-commit.sh (hunk 1 above)
Comment (nilsdeppe) [reply tools/Hooks/pre-commit.sh:53]: I think we could reasonably just run it if we can add the changes directly to the commit changes. I haven't looked into this.

@wthrowe how would you feel about this?

**Review by nilsdeppe — DISMISSED** [review]
I'm happy with this. Thanks for making these changes! This would've been enough to install black and get formatting right :)

File: tools/Hooks/pre-commit.sh (hunk 1 above)
Comment (wthrowe) [reply tools/Hooks/pre-commit.sh:53]: Doing the same thing we do with clang-format makes sense.  I am agnostic on aborting the commit if the check fails.  I'm opposed to applying the changes during the commit process.

Comment (nilsdeppe) [reply tools/Hooks/pre-commit.sh:53]: I agree, having something change code for you usually causes more challenges than it solves, especially in a case like this where the thing you need to do is 7 characters

**Comment by nilsdeppe** [comment]
> What do you think about also doing this
> 
> ```diff
> diff --git a/docs/Installation/Installation.md b/docs/Installation/Installation.md
> index 014cef7b70..b89fb99938 100644
> --- a/docs/Installation/Installation.md
> +++ b/docs/Installation/Installation.md
> @@ -93,7 +93,7 @@ all of these dependencies.
>    Alternatively, you can set `BOOTSTRAP_PY_DEPS=ON` when configuring a build
>    with CMake to install missing Python packages into the build directory
>    automatically.
> -  <details>
> +  <details open>
>    \include support/Python/requirements.txt
>    </details>
>  
> @@ -107,7 +107,7 @@ all of these dependencies.
>    Install with `pip3 install -r support/Python/dev_requirements.txt`.
>    Make sure you are working in a [Python venv](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment)
>    before installing packages.
> -  <details>
> +  <details open>
>    \include support/Python/dev_requirements.txt
>    </details>
>  * [Google Benchmark](https://github.com/google/benchmark) - to do
> ```
> 
> When I searched for hidden things on Chrome (linux), it actually opened the `Details` tab and highlighted the search. However, after some searching, apparently this behavior is browser dependent and not all browsers will search hidden content. So if we want to guarantee searchability in the installation docs, we can just have the details open initially (or just get rid of the hidden details and have the file included always).

I like this suggestion. @nilsvu would you be okay adding that?

**Comment by nilsvu** [comment]
Well I put the requirements.txt file in `<details>` for a reason. You don't really need to know the list of packages, just `pip install -r [dev_]requirements.txt`. IMO showing the `<details>` is unnecessarily distracting on the installation page.
I added another hint how to install black and isort to the "running the checks locally" docs.

**Comment by nilsdeppe** [comment]
But the point remains, if I search the installation page for any of our python dependencies they may or may not show up...

**Review by nilsdeppe — APPROVED** [review]
I'd rather get these improvements and worry about whether to detail or not later...

**Pull request merged**